From e4939f253dac5665e3a21626f9f18da02a73f8a3 Mon Sep 17 00:00:00 2001
From: t0m <bobtfish@bobtfish.net>
Date: Thu, 10 Jun 2010 20:34:49 +0100
Subject: [PATCH] Skip test when appropriate. Fixes RT#58307

---
 t/08-dbic-mappedfields.t |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/t/08-dbic-mappedfields.t b/t/08-dbic-mappedfields.t
index a3f85f4..2c2fbc1 100644
--- a/t/08-dbic-mappedfields.t
+++ b/t/08-dbic-mappedfields.t
@@ -11,11 +11,16 @@ BEGIN {
     my @needed = qw/
         Catalyst::Model::DBIC::Schema
         Catalyst::Authentication::Store::DBIx::Class
+        DBIx::Class::Optional::Dependencies
     /;
     plan skip_all => "One of the required classes for this test $@ (" . join(',', @needed) . ") not found."
         unless eval {
             Class::MOP::load_class($_) for @needed; 1;
         };
+    require DBIx::Class;
+
+    plan skip_all => 'Test needs ' . DBIx::Class::Optional::Dependencies->req_missing_for('admin')
+        unless DBIx::Class::Optional::Dependencies->req_ok_for('admin');
 }
 
 use Catalyst::Test qw/TestAppDBIC/;
-- 
1.6.2.1

